File: sdks/python/examples/rag_query.py
"""Search a knowledge base."""

import asyncio

from universal_agent_runtime_sdk import Client


async def main() -> None:
    async with Client("http://localhost:1906") as client:
        result = await client.search("default", "What is UAR?")
        for match in result.results:
            print(match.score, match.content)


if __name__ == "__main__":
    asyncio.run(main())

File: sdks/rust/examples/rag_query.rs
use universal_agent_runtime_sdk::Client;
#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new(
        std::env::var("UAR_BASE_URL").unwrap_or_else(|_| "http://localhost:1906".into()),
    )?;
    let kb_id = std::env::var("UAR_KB_ID").unwrap_or_else(|_| "replace-me".into());
    for hit in client
        .knowledge()
        .search(&kb_id, "What does this knowledge base contain?")
        .await?
        .results
    {
        println!("{:.3}: {}", hit.score, hit.content);
    }
    Ok(())
}
